custom/icds_reports/views.py

```diff
@@ -1949,3 +1949,40 @@ def get_type_code(self, data_type):
             "awc": 3
         }
         return type_map[data_type]
+
+
+@location_safe
+@method_decorator([api_auth, toggles.mwcd_indicators.required_decorator()], name='dispatch')
+class MWCDDataView(View):
+
+    def message(self, message_name):
+        error_messages = {
+            "unknown_error": "Unknown Error occured",
+            "no_data": "Data does not exists"
+        }
+
+        return error_messages[message_name]
+
+    def get(self, request, *args, **kwargs):
+
+        try:
+            data = get_mwcd_indicator_api_data()
+            response = {'isSuccess': True,
+                        'message': 'Data Sent Successfully',
+                        'Result': {
+                            'response': data
+                        }}
+            return JsonResponse(response)
+        except NICIndicatorsView.DoesNotExist:
```

Comment:
its weird to use an exception from another view. May be better to pull this out into its own exception and use it in both rather than the propery of that view